### DISCUSSION TITLE
Binding failed

### DISCUSSION THREAD

**Discussion by OrangeANDc** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3229] [Q&A] [answered]
Hello，my develop environment is framework4.72+ prism。
when i use prism‘s InitializeShell method to start my loginView，visual studio show the following errors，then after logined，mainview’s togglebutton cannot function properity。
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/41464614/be70cb5d-bf86-40db-aa08-371c9b6409c3)

![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/41464614/279da258-4ef7-4d2b-b8c1-e002d8e53d28)

but if i don‘t use  InitializeShell  method，displaye the mainview  directly，This problem would not apperance。Do you konw how to handel？

**Comment by Keboo** [comment, marked as answer]
@OrangeANDc that looks like it is expecting your UI to also have a ToggleButton named `MenuToggleButton`. Does that element exist in your MainWindow?

**Reply by OrangeANDc** [reply]
that is strange when i change this way to bind then it works， 
![image](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/assets/41464614/5a261bf6-93f0-4954-9e23-da2dfd038fd4)
I find it in Microsoft official documents:
In WPF and XAML 2006, element references are addressed by the framework-level feature of ElementName binding. For most WPF applications and scenarios, ElementName binding should still be used. Exceptions to this general guidance might include cases where there are data context or other scoping considerations that make data binding impractical and where markup compilation is not involved.
